src/components/transactions/history/List.tsx

```diff
@@ -81,81 +87,60 @@ export function List({ setIsScrolled, targetRef }: Props) {
+                    ? originalWalletBridgeTransactionIndex
+                    : depracatedWalletTransactionIndex;
+
+            // Don't process if we can't find the transaction in the wallet transactions
+            if (walletBridgeTransactionIndex < 0) return;
+
+            extendedTransactions[walletBridgeTransactionIndex] = {
+                ...extendedTransactions[walletBridgeTransactionIndex],
+                bridgeTransactionDetails: {
+                    status: bridgeTx.status,
+                    transactionHash: bridgeTx.transactionHash,
+                    amountAfterFee: bridgeTx.amountAfterFee,
+                },
+            };
+        });
 
-        return transactions.sort((a, b) => getTimestampFromTransaction(b) - getTimestampFromTransaction(a));
-    }, [baseTx, bridgeTransactions, tx_history_filter]);
-
-    const adjustedTransactions = useMemo(() => {
-        return combinedTransactions.reduce(
-            (acc, tx, index) => {
-                if (isBridgeTransaction(tx) && index > 0) {
-                    const previousTransactions = acc.slice(-5); // Get up to the last 5 transactions
-                    const matchingTransaction = previousTransactions.find(
-                        (prevTx) =>
-                            isTransactionInfo(prevTx) &&
-                            prevTx.amount === Number(tx.tokenAmount) &&
-                            prevTx.payment_id === tx.paymentId &&
-                            prevTx.dest_address === coldWalletAddress
-                    );
-
-                    if (matchingTransaction) {
-                        const removedBridgeTransaction = acc.splice(acc.indexOf(matchingTransaction), 1)[0];
-                        if (removedBridgeTransaction && isTransactionInfo(removedBridgeTransaction)) {
-                            const updatedTransaction: BackendBridgeTransaction = {
-                                sourceAddress: removedBridgeTransaction.source_address,
-                                destinationAddress: tx.destinationAddress,
-                                status: tx.status,
-                                createdAt: tx.createdAt,
-                                tokenAmount: tx.tokenAmount,
-                                amountAfterFee: tx.amountAfterFee,
-                                feeAmount: tx.feeAmount,
-                                paymentId: tx.paymentId,
-                                mined_in_block_height: removedBridgeTransaction.mined_in_block_height,
-                            };
-                            acc.push(updatedTransaction);
-                            return acc;
-                        }
-                    }
-                }
-                acc.push(tx);
-                return acc;
-            },
-            [] as (TransactionInfo | UserTransactionDTO)[]
-        );
-    }, [combinedTransactions, coldWalletAddress]);
+        return extendedTransactions;
+    }, [baseTx, bridgeTransactions]);
```

Comment:
in understand the other dependency missing, but is `coldWalletAddress` intentionally left out for this?